### PR TITLE
Add support for per-release URLs

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: libappstream-glib/as-release.c [hunk 1]
```diff
@@ -1019,6 +1097,17 @@ as_release_node_parse_dep11 (AsRelease *release, GNode *node,
 			}
 			continue;
 		}
+		if (g_strcmp0 (tmp, "Url") == 0) {
```
Comment (ximion) [inline libappstream-glib/as-release.c]: I'm quite sure that's a lower-cased `url` - in the YAML, only the top nodes are upper-cased, every child node is lower-cased all the way.

Comment (hughsie) [reply libappstream-glib/as-release.c]: God catch, thanks.

File: libappstream-glib/as-release.c [hunk 2]
```diff
@@ -769,6 +774,67 @@ as_release_set_description (AsRelease *release,
...
+ * @release: a #AsRelease instance.
+ *
+ * Gets the URLs set for the application.
+ *
+ * Returns: (transfer none): hash table of URLs
+ *
+ * Since: 0.7.15
+ **/
+GHashTable *
+as_release_get_urls (AsRelease *release)
```
Comment (kalev) [inline libappstream-glib/as-release.c]: Is this actually needed as public API? Seems to leak a lot of internal details (hash table).

File: libappstream-glib/as-release.c [hunk 3]
```diff
@@ -769,6 +774,67 @@ as_release_set_description (AsRelease *release,
...
+ * @url_kind: the URL kind, e.g. %AS_URL_KIND_HOMEPAGE.
+ *
+ * Gets a URL.
+ *
+ * Returns: string, or %NULL if unset
+ *
+ * Since: 0.7.15
+ **/
+const gchar *
+as_release_get_url_item (AsRelease *release, AsUrlKind url_kind)
```
Comment (kalev) [inline libappstream-glib/as-release.c]: Maybe call it just as_release_get_url() ? _item at the end seems unnecessary and just makes it more of a mouthful.

File: libappstream-glib/as-release.c (hunk 2 above)
Comment (hughsie) [reply libappstream-glib/as-release.c]: I just coped the design (mistake?) from AsApp. I can nuke this, I'm not using it at the moment. Ideas welcome.

File: libappstream-glib/as-release.c (hunk 3 above)
Comment (hughsie) [reply libappstream-glib/as-release.c]: I just coped the name (another mistake?) from AsApp. Maybe `as_release_get_url_by_kind`?

File: libappstream-glib/as-release.c (hunk 2 above)
Comment (kalev) [reply libappstream-glib/as-release.c]: I think I'd leave it out then and add it only when something actually needs it.

File: libappstream-glib/as-release.c (hunk 3 above)
Comment (kalev) [reply libappstream-glib/as-release.c]: Sure, works for me!

Comment (ximion) [reply libappstream-glib/as-release.c]: libappstream just calls this `as_release_get_url(rel, kind)` and has a `as_release_get_url(rel, kind, url)`

File: libappstream-glib/as-enums.h [hunk 4]
```diff
@@ -84,6 +85,7 @@ typedef enum {
 	AS_URL_KIND_HELP,		/* Since: 0.1.5 */
 	AS_URL_KIND_MISSING,		/* Since: 0.2.2 */
 	AS_URL_KIND_TRANSLATE,		/* Since: 0.6.1 */
+	AS_URL_KIND_DETAILS,		/* Since: 0.7.15 */
```
Comment (ximion) [inline libappstream-glib/as-enums.h]: In libappstream I created a `AS_RELEASE_URL_KIND` to distinguish between what URLs are valid in releases and which ones aren't.
That distinction is probably not necessary though.

File: libappstream-glib/as-release.c (hunk 3 above)
Comment (kalev) [reply libappstream-glib/as-release.c]: I think as_release_get_url(rel, kind) would be my preference as well, but as_release_get_url_by_kind() works as well.

Comment (hughsie) [reply libappstream-glib/as-release.c]: as_release_get_url it is then. thanks.

File: libappstream-glib/as-release.c (hunk 2 above)
Comment (hughsie) [reply libappstream-glib/as-release.c]: Okay, agree. Thanks.

**Comment by kalev** [comment]
Looks good to me!

**Pull request merged**